sigstore-java/src/main/java/dev/sigstore/encryption/Keys.java

```diff
@@ -60,48 +62,26 @@ public class Keys {
    */
   public static PublicKey parsePublicKey(byte[] keyBytes)
       throws InvalidKeySpecException, IOException, NoSuchAlgorithmException {
-    PemReader pemReader =
-        new PemReader(
-            new InputStreamReader(new ByteArrayInputStream(keyBytes), StandardCharsets.UTF_8));
-    PemObject section = null;
-    try {
-      section = pemReader.readPemObject();
-      if (pemReader.readPemObject() != null) {
+    try (PEMParser pemParser =
+        new PEMParser(
+            new InputStreamReader(new ByteArrayInputStream(keyBytes), StandardCharsets.UTF_8))) {
+      var keyObj = pemParser.readObject(); // throws DecoderException
+      if (keyObj == null) {
         throw new InvalidKeySpecException(
             "sigstore public keys must be only a single PEM encoded public key");
       }
+      JcaPEMKeyConverter converter = new JcaPEMKeyConverter();
+      if (keyObj instanceof SubjectPublicKeyInfo) {
+        PublicKey pk = converter.getPublicKey((SubjectPublicKeyInfo) keyObj);
+        if (!List.of("ECDSA", "EC", "RSA", "Ed25519", "EdDSA").contains(pk.getAlgorithm())) {
```

Comment:
do we not have a static set of these defined somewhere?

Reply:
Yeah I can make that a bit more reusable